WordPress/src/main/java/org/wordpress/android/util/analytics/AnalyticsTrackerWrapper.kt

```diff
@@ -38,8 +38,13 @@ class AnalyticsTrackerWrapper
         AnalyticsTracker.track(stat, properties + feature.toParams())
     }
 
-    fun track(stat: Stat, site: SiteModel?) {
-        AnalyticsUtils.trackWithSiteDetails(stat, site)
+    @JvmOverloads
+    fun track(stat: Stat, site: SiteModel?, properties: Map<String, Any?>? = null) {
+        if (properties == null) {
+            AnalyticsUtils.trackWithSiteDetails(stat, site)
```

Comment:
Continuing our [discussion here](https://github.com/wordpress-mobile/WordPress-Android/pull/15465#discussion_r741257414) wdyt of removing the `if` clause completely and use in all cases the call:
`AnalyticsUtils.trackWithSiteDetails(this, stat, site, properties)`?
The [`AnalyticsUtils.trackWithSiteDetails(stat, site)`](https://github.com/wordpress-mobile/WordPress-Android/blob/25e4dd4cb2250ee9a9dd06e9069a558a2258abb8/WordPress/src/main/java/org/wordpress/android/util/analytics/AnalyticsUtils.java#L232) calls:
[`trackWithSiteDetails(stat, site, null)`](https://github.com/wordpress-mobile/WordPress-Android/blob/25e4dd4cb2250ee9a9dd06e9069a558a2258abb8/WordPress/src/main/java/org/wordpress/android/util/analytics/AnalyticsUtils.java#L281) and instantiates an `AnalyticsTrackerWrapper`.

Reply:
Good point and makes sense! Besides, `trackWithSiteDetails(..., properties)` checks for null properties anyway so, it covers the conditional 👍. Done with 5c34afa.

Reply:
Thank you for all the changes Stefanos 🙇 
I don't have any further comments on this PR 🎉

Reply:
Grand! Thanks for the reviews and keen eye Antonis! I will assign the PR to myself and go ahead with the merge 👍